### ISSUE TITLE
[Bug][Dynamic Embedding] improper optimizier state_dict `momentum2` key while constructing PSCollection

### ISSUE THREAD

**Issue by JacoCheung** [meta-pytorch/torchrec#2177]
# Describe the bug

A PSCollection should contain optimizer states besides weights. The optimizer states tensors are obtained directly [from EmbeddingCollection Module](https://github.com/pytorch/torchrec/blob/main/contrib/dynamic_embedding/src/torchrec_dynamic_embedding/ps.py#L165-L168). 

However, the [sharded_module.fused_optimizer.state_dict()['state']](https://github.com/pytorch/torchrec/blob/main/contrib/dynamic_embedding/src/torchrec_dynamic_embedding/ps.py#L153C32-L153C75) does not contain key `{table_name}.momentum2` because

1. [TBE::get_optimizer_state()](https://github.com/pytorch/FBGEMM/blob/main/fbgemm_gpu/fbgemm_gpu/split_table_batched_embeddings_ops_training.py#L1860-L1885) which is used by `PSCollection` will not return key like `xxx.momentum1` or `xxx.momentum2`.  They are customized by TBE. 
2. The states keys are renamed by [torchrec::EmbeddingFusedOptimizer](https://github.com/pytorch/torchrec/blob/v0.6.0-rc2/torchrec/distributed/batched_embedding_kernel.py#L349-L354). The first state falls back on `xxx.momentum1` while the left keys are copied from above retrived results.

See the below illustration where optimizer is Adam. The expected number of state tensors should be 2, but the it eventually gives `momentum1` and leaves `momentum2` (which is synonymously `exp_avg_sq`) out. 

![opt report](https://github.com/pytorch/torchrec/assets/32166257/42e99d26-8d3a-486a-be62-cae534423b8b)

It will pose impact on all kinds of optimizer that contains momentum2.

**Comment by iamzainhuda** [comment]
Yup this shouldn't be intended behaviour. Feel free to submit a PR I can go ahead and review.
